eessi_bot_job_manager.py

```diff
@@ -48,185 +41,277 @@ def mkdir(path):
+                        log(
+                            "get_known_jobs(): entry %s in %s"
+                            " is not recognised as a symlink"
+                            % (full_path, self.submitted_jobs_dir),
+                            self.logfile,
+                        )
                 else:
-                    log("get_known_jobs(): entry %s in %s doesn't match regex" % (fname,self.submitted_jobs_dir), self.logfile)
+                    log(
+                        "get_known_jobs(): entry %s in %s "
+                        "doesn't match regex" % (fname, self.submitted_jobs_dir),
+                        self.logfile,
+                    )
         else:
-            log("get_known_jobs(): directory '%s' does not exist -> assuming no jobs known previously" % self.submitted_jobs_dir, self.logfile)
+            log(
+                "get_known_jobs(): directory '%s' "
+                "does not exist -> assuming no jobs known previously"
+                % self.submitted_jobs_dir,
+                self.logfile,
+            )
 
         return known_jobs
 
-
-    #new_jobs = job.manager.determine_new_jobs(known_jobs, current_jobs)
+    # new_jobs = job.manager.determine_new_jobs(known_jobs, current_jobs)
     def determine_new_jobs(self, known_jobs, current_jobs):
         # known_jobs is a dictionary: jobid -> {'jobid':jobid}
-        # current_jobs is a dictionary: jobid -> {'jobid':jobid,'state':val,'reason':val}
+        # current_jobs is a dictionary: jobid -> {'jobid':jobid,
+        #                                         'state':val,'reason':val}
         new_jobs = []
         for ckey in current_jobs:
             if not ckey in known_jobs:
                 new_jobs.append(ckey)
 
         return new_jobs
 
-
-    #finished_jobs = job.manager.determine_finished_jobs(known_jobs, current_jobs)
+    # finished_jobs = job.manager.determine_finished_jobs(known_jobs,
+    #                                                     current_jobs)
     def determine_finished_jobs(self, known_jobs, current_jobs):
         # known_jobs is a dictionary: jobid -> {'jobid':jobid}
-        # current_jobs is a dictionary: jobid -> {'jobid':jobid,'state':val,'reason':val}
+        # current_jobs is a dictionary: jobid -> {'jobid':jobid,
+        #                                         'state':val,'reason':val}
         finished_jobs = []
         for kkey in known_jobs:
             if not kkey in current_jobs:
                 finished_jobs.append(kkey)
 
         return finished_jobs
 
+    def read_job_pr_metadata(self, job_metadata_path):
+        # check if metadata file exist
+        isfile = os.path.isfile(job_metadata_path)
+        log("The test for metafile returns: %s" % isfile)
+        if isfile is False:
+            log("this is a non bot job and it can't be processed")
+            return None
```

Comment:
This function should also actually *read* the metadata file, if it exists (see `metadata.read(job_metadata_path)` below)

Reply:
So this function should do 3 things:

* check if the file exists, return `None` if not
* if the file exists, read it, extract the `PR` section, and return that as a value